src/app/trajectory/map/map.page.ts

```diff
@@ -116,18 +118,31 @@ export class MapPage implements OnInit, OnDestroy {
 
     // TODO: rework this with optional inference type parameter,
     //   which we subscribe to and use to set zoom & open popup
-    if (history.state.center)
+    if (history.state.center) {
       this.mapBounds = latLng(history.state.center).toBounds(100)
+    }
   }
 
   onMapReady(map: Map) {
     this.map = map
   }
 
+  onMapMoved(map: Map) {
+    if (!this.isTogglingFollowPosition) {
+      this.followPosition = false
+    } else {
+      this.isTogglingFollowPosition = false
+    }
+  }
+
   onToggleFollowMode() {
+    this.isTogglingFollowPosition = true
     this.followPosition = !this.followPosition
-    if (this.followPosition)
+    if (this.followPosition) {
       this.mapBounds = this.lastLocation.getLatLng().toBounds(100)
+    } else {
+      this.mapBounds = this.polyline.getBounds()
```

Comment:
I'd leave that as is; when I disable follow mode, it doesn't mean that I want to see the whole map. Thats behaviour I hate in other apps :D:

Reply:
Interesting - for me that would be expected and desireable behaviour. 😄  I removed it for now.